src/chains.rs

```diff
@@ -489,6 +494,12 @@ struct ChainFormatterShared<'a> {
     // The number of children in the chain. This is not equal to `self.children.len()`
     // because `self.children` will change size as we process the chain.
     child_count: usize,
+    // Whether the parent element of the chain exceeds the max width
+    has_long_parent: bool,
+    // Whether one or more of the inner chain items exceed the max width
+    has_long_inner_item: bool,
+    // Whether the final chain item exceeds the max width
+    has_long_tail: bool,
```

Comment:
instead of a single `has_long` flag, these fields are used to have a more detailed view of where the chain has a long element based on how chains are formatted (parent and tail are formatted separately)